src/Core/Tool/Authorizer/TosAuthorizer.php

```diff
@@ -39,9 +39,14 @@ class TosAuthorizer implements Authorizer
 
     /* Authorizer */
     public function isAllowed(Entity $entity, $privilege)
-    {
+    {   
+        // Get the current user ID
+        $current_user = service('session.user');
```

Comment:
This is actually the same thing returned by `getUser` below. I think you need to compare with the $entity->user_id